cl/search/models.py

```diff
@@ -2158,6 +2161,80 @@ def has_private_authority(self):
             self._has_private_authority = private
         return self._has_private_authority
 
+    @property
+    def authorities_with_data(self):
+        """Returns a list of dictionaries containing useful data on every
+        authority, for eventual injection into a view template. Wrapping each
+        authority in such a dictionary is necessary because we have to
+        calculate and append an extra data field (relating to citation counts)
+        on-the-fly.
+        The returned list is sorted by that citation count field.
+        """
+        authorities_with_data = [
+            {
+                "caption": authority.caption,
+                "docket": authority.docket,
+                "date_filed": authority.date_filed,
+                "absolute_url": authority.get_absolute_url,
+                "sub_opinions": authority.sub_opinions,
+                # Number of other opinions that cite this authority
+                "unique_citation_count": authority.citation_count,
+                # Number of citations from this cluster to this authority
+                "this_cluster_citation_count": get_citation_depth_for_cluster_pks(
+                    citing_cluster_pk=self.pk, cited_cluster_pk=authority.pk,
+                ),
+            }
+            for authority in self.authorities
+        ]
```

Comment:
I'm guessing you tried this, but can't you cast a QuerySet to a list and then just add this attribute to each item, then sort the list? I'm surprised that's not possible, I guess, and I suspect it'd be a lot tidier?

Reply:
Ah, for some reason I thought it was not possible to add a new attribute to an object that had already been instantiated, so I thought I needed to wrap each `OpinionCluster` object in a dictionary! But thank god for dynamic typing. This is much simplified now: 2ffe8c7

Reply:
Calling this `cluster_references` is a little confusing, isn't it? Why not just call it `depth` or something?

Reply:
👍